### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by Saverio976**
merge en RÉU

### CODE REVIEW

**Review by Beafowl-Pull — APPROVED** [review]

File: include/Arcade/Core/IGameModule.hpp [hunk 1]
```diff
@@ -45,8 +54,22 @@ namespace Arcade {
...
+                 *receive the event CHANGE_GAME, called when there is not param
+                 *(in the core loop)
                  */
                 virtual void changeGame() = 0;
+                /**
+                 * @brief Get Game Event Only Manager
+                 *
+                 * @return The event manager only used for game
+                 */
+                virtual Arcade::ECS::IEventManager &getGameEventManager() = 0;
```
Comment (romainpanno) [inline include/Arcade/Core/IGameModule.hpp:72]: shared ptr instead of reference

File: include/Arcade/ECS/IEntity.hpp [hunk 2]
```diff
@@ -44,7 +44,7 @@ namespace Arcade {
                  * @return The list of components filtered
```
Comment (romainpanno) [inline include/Arcade/ECS/IEntity.hpp:44]: add doc for .at()

File: include/Arcade/Game/IScene.hpp [hunk 3]
```diff
@@ -41,13 +41,13 @@ namespace Arcade {
                  *
                  * @return The entity manager of the scene
                  */
-                virtual ECS::IEntityManager &getEntityManager() = 0;
+                virtual std::unique_ptr<ECS::IEntityManager> &getEntityManager() = 0;
```
Comment (romainpanno) [inline include/Arcade/Game/IScene.hpp:44]: shared instead of unique reference

File: include/Arcade/Game/IScene.hpp [hunk 4]
```diff
@@ -41,13 +41,13 @@ namespace Arcade {
...
                 /**
                  * @brief Get the system manager of the scene
                  *
                  * @return The system manager of the scene
                  */
-                virtual ECS::ISystemManager &getSystemManager() = 0;
+                virtual std::unique_ptr<ECS::ISystemManager> &getSystemManager() = 0;
```
Comment (romainpanno) [inline include/Arcade/Game/IScene.hpp:50]: shared instead of unique reference

**Review by romainpanno — CHANGES_REQUESTED** [review]

**Review by KitetsuK — APPROVED** [review]

**Pull request merged**